File: app/src/main/java/com/comfortable/eyes/SharedTimeState.java
package com.comfortable.eyes;

import android.content.Context;
import android.content.SharedPreferences;

public class SharedTimeState {
    private Context mContext;
    private SharedPreferences pref;
    private SharedPreferences.Editor edit;
    private Time time;
    private String currentDate;

    public SharedTimeState(Context context) {
        this.mContext = context;
        time = new Time();
        pref = mContext.getSharedPreferences("SharedState", 0);
        edit = pref.edit();
        currentDate = pref.getString("currentDate", "0");
        time.hour = pref.getInt("hour", 0);
        time.minutes = pref.getInt("minutes", 0);
        time.seconds = pref.getInt("seconds", 0);
    }

    public void setTime(Time t) {
        edit.putInt("hour", t.hour);
        edit.putInt("minutes", t.minutes);
        edit.putInt("seconds", t.seconds);
        edit.commit();
    }

    public void setCurrentDate(String d) {
        edit.putString("currentDate", d);
        edit.commit();
    }

    public void resetTime(Time t) {
        t.hour = 0;
        t.minutes = 0;
        t.seconds = 0;
        setTime(t);
    }

    public Time getTime() {
        return this.time;
    }

    public String getCurrentDate() {
        return this.currentDate;
    }
}

File: app/src/main/java/com/comfortable/eyes/state/SharedTimeState.kt
package com.comfortable.eyes.state

import com.comfortable.eyes.AppContext.Companion.context
import android.os.SystemClock

object SharedTimeState {
    private val pref = context.getSharedPreferences("SharedTimeState", 0)
    private val edit = pref.edit()

    var startTime: Long = 0
    var usedTime: Long = pref.getLong("time_used", 0)
    var dayOfYear: Int = pref.getInt("day_of_year", 0)

    fun init() {
        usedTime = 0
        startTime = SystemClock.elapsedRealtime()
    }

    fun commitState() {
        edit.putLong("time_used", usedTime)
        edit.putInt("day_of_year", dayOfYear)
        edit.commit()
    }
}